tests/test_datastore.py

```diff
@@ -0,0 +1,224 @@
+from base import TestBase, unittest
+from mock import Mock, sentinel, patch, MagicMock
+from threading import Lock
+import numbers
+
+from virtwho.datastore import Datastore
+
+
+class ArbitraryTestItem(object):
+    def __init__(self, x, y, z):
+        self.x = x
+        self.y = y
+        self.z = z
+
+class TestDatastore(TestBase):
+
+    def mock_copy(self, item):
+        # Return the item with some predictable modification
+        # Used to prove that the result of copy method is used in the right
+        # places
+        return item, item
+
+    def assert_equivalent(self, item1, item2):
```

Comment:
Self comment, this might be absurd. I could be convinced to trust that any object we'd like to test equivalence of has a sane __eq__ implementation method.